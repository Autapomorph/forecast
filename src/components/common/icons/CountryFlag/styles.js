import styled, { css } from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledCountryFlag = styled.span`
  display: inline-block;
  position: relative;
  width: 1.33333333em;
  background-image: ${({ country }) => `url(/images/countries/flags/4x3/${country}.svg)`};
  background-position: 50%;
  background-size: contain;
  background-repeat: no-repeat;

  ::before {
    content: '\00a0';
  }

  ${
    /* sc-dec */ ({ background }) =>
      background &&
      css`
        background-position: 50%;
        background-size: contain;
        background-repeat: no-repeat;
      `
  }

  ${
    /* sc-dec */ ({ squared }) =>
      squared &&
      css`
        width: 1em;
        background-image: ${({ country }) => `url(/images/countries/flags/1x1/${country}.svg)`};
      `
  }

  ${
    /* sc-dec */ ({ size }) =>
      size &&
      css`
        font-size: ${size};
      `
  }
`;
