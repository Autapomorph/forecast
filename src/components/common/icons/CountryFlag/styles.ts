import styled from 'styled-components/macro';

type Props = {
  country: string;
};

export const CountryFlag = styled.span`
  display: inline-block;
  width: 1em;
  height: 1em;
  border-radius: 50%;
  vertical-align: middle;
  background-image: ${({ country }: Props) => `url(/images/countries/flags/${country}.svg) `};
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
`;
