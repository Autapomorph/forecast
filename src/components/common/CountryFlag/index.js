import React from 'react';

import { StyledCountryFlag } from './styles';

const CountryFlag = props => <StyledCountryFlag {...props} />;

export default CountryFlag;
