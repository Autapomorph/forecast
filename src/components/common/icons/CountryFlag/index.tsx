import React from 'react';

import * as S from './styles';

type Props = {
  country: string;
  style?: React.CSSProperties;
};

const CountryFlag = ({ country, style }: Props): React.ReactElement => (
  <S.CountryFlag country={country} style={style} />
);

export default CountryFlag;
