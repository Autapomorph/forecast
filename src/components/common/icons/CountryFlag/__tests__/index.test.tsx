import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CountryFlag from '..';

describe('CountryFlag tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<CountryFlag country="ru" />);

    expect(wrapper).toMatchSnapshot();
  });

  xit('should render country as background-image', () => {
    const wrapper = shallow(<CountryFlag country="ru" />);

    expect(wrapper).toMatchSnapshot();
    // @ts-ignore
    expect(toJson(wrapper)).toHaveStyleRule('background-image', expect.stringContaining('ru'));
  });
});
