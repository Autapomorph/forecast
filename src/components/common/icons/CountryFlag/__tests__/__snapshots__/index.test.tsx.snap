// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryFlag tests renders properly 1`] = `
<styles__CountryFlag
  country="ru"
/>
`;
